modules/cas_cache/ocf_env.c

```diff
@@ -102,6 +104,8 @@ env_allocator *env_allocator_create(uint32_t size, const char *name)
 		goto err;
 	}
 
+	BUG_ON(num_online_cpus() >= 1 << CAS_RPOOL_MAX_CPU_BITS);
```

Comment:
This should be `>`, not `>=`. We also may want to perform this sanity check at the very beginning of this function. Nevertheless, taking into account that you just check if machine has more than over 3 billion (!) CPUs, it seems to be well overdone.

Reply:
<insert Bill Gates 640K quote>

Reply:
This BUG_ON is in case someone adds extra fields to this bitfield and lowers CAS_RPOOL_MAX_CPU_BITS

Reply:
Also I think > is ok

Reply:
This function is called only once during initialization, so impact is negligible

Reply:
I neeed to put this check after variables initialization. And this check for NULL allocator looks good close to allocator initialization. I prefer it this way, but I can change this if you insist.